src/main/java/org/joshsim/lang/interpret/visitor/JoshConfigDiscoveryVisitor.java

```diff
@@ -0,0 +1,136 @@
+package org.joshsim.lang.interpret.visitor;
+
+import java.util.HashSet;
+import java.util.Set;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.joshsim.engine.config.DiscoveredConfigVar;
+import org.joshsim.lang.antlr.JoshLangBaseVisitor;
+import org.joshsim.lang.antlr.JoshLangParser;
+
+/**
+ * Visitor to discover configuration variables used in Josh scripts.
+ *
+ * <p>This visitor traverses the AST and collects all configuration variable
+ * references found in config expressions. For example, "config example.testVar"
+ * would add "example.testVar" to the discovered set, while "config example.testVar else 5 m"
+ * would add "example.testVar(5 m)" to show the default value.</p>
+ */
+public class JoshConfigDiscoveryVisitor extends JoshLangBaseVisitor<Set<DiscoveredConfigVar>> {
+
+  /**
+   * Creates a new config discovery visitor.
+   */
+  public JoshConfigDiscoveryVisitor() {
+    super();
+  }
+
+  @Override
+  public Set<DiscoveredConfigVar> visitConfigValue(JoshLangParser.ConfigValueContext ctx) {
+    String variableName = ctx.name.getText();
+    DiscoveredConfigVar discovered = new DiscoveredConfigVar(variableName);
+    Set<DiscoveredConfigVar> result = new HashSet<>();
+    result.add(discovered);
+
+    // Continue visiting children to collect any nested config values
+    Set<DiscoveredConfigVar> childResult = visitChildren(ctx);
+    if (childResult != null) {
+      result.addAll(childResult);
+    }
+
+    return result;
+  }
+
+  @Override
+  public Set<DiscoveredConfigVar> visitConfigValueWithDefault(
+      JoshLangParser.ConfigValueWithDefaultContext ctx) {
+    String variableName = ctx.name.getText();
+    String defaultValueString = extractDefaultValueString(ctx.defaultValue);
+    DiscoveredConfigVar discovered = new DiscoveredConfigVar(variableName, defaultValueString);
+    Set<DiscoveredConfigVar> result = new HashSet<>();
+    result.add(discovered);
+
+    // Continue visiting children to collect any nested config values
+    Set<DiscoveredConfigVar> childResult = visitChildren(ctx);
+    if (childResult != null) {
+      result.addAll(childResult);
+    }
+
+    return result;
+  }
+
+  /**
+   * Extracts a readable string representation from a default value expression.
+   *
+   * <p>This method traverses the parse tree to build a human-readable string
+   * representation of the default value, handling common cases like numbers
+   * with units, strings, and simple expressions.</p>
+   *
+   * @param defaultValueContext The parse tree context for the default value expression
+   * @return A string representation of the default value
+   */
+  private String extractDefaultValueString(ParseTree defaultValueContext) {
+    if (defaultValueContext == null) {
+      return "";
+    }
+
+    // Handle simple cases based on context type using new-style switch
+    switch (defaultValueContext.getClass().getSimpleName()) {
```

Comment:
Is there a way we can have the defaultValueContext accept us with visitor instead of this switch case on class name? We should avoid inspection if possible. Let's review the other ANTLR visitors in the project.